### PR TITLE
openqa-clone-job: Add support for full test URLs

### PR DESCRIPTION
**Pull request by okurz**
Instead of needing the job ID split out we can also specify full test
URLs which makes copy-paste easier :)

### CODE REVIEW

**Comment by okurz** [comment]
hm, that diff will be hard to read. I guess I will create another PR first: https://github.com/os-autoinst/openQA/pull/2138

**Comment by okurz** [comment]
rebased, the diff is much easier to read now.

**Comment by okurz** [comment]
tidy consistently wants to patch the clone_job.pl script which is just a symlink and therefore replacing back the symlink with a duplicated copy :( Anyone has an idea how to prevent that?

**Comment by okurz** [comment]
solved with https://github.com/os-autoinst/openQA/pull/2137/commits/0ad31d824838a8a06480af5fdc849cd7946f451e :)

File: script/tidy [hunk 1]
```diff
@@ -46,7 +46,7 @@ find . -name '*.tdy' -delete
 # .pc directory is used for "quilt" patch system (in Debian/Ubuntu)
 # it contains pre-patched files and should be ignored
 # shellcheck disable=SC2207
-files=($(find . ! -path '*/.pc/*' \( -name '*.p[lm]' -o -name '*.t' \)))
+files=($(find . ! -path '*/.pc/*' -type f \( -name '*.p[lm]' -o -name '*.t' \)) $(file --mime-type script/* | sed -n 's/^\(.*\):.*text\/x-perl.*$/\1/p'))
```
Comment (kraih) [inline script/tidy]: This change is unrelated to the rest of the PR.

Comment (okurz) [reply script/tidy]: Actually there is a clear relation: https://github.com/os-autoinst/openQA/pull/2137#issuecomment-507600748 explains that. However I can create another PR if this is what you prefer: https://github.com/os-autoinst/openQA/pull/2155

File: script/openqa-clone-job [hunk 2]
```diff
@@ -135,21 +145,38 @@ sub usage($) {
     }
 }
 
+sub split_jobid {
+    my ($url_string) = @_;
+    my $url          = Mojo::URL->new($url_string);
+    my $host_url     = $url->scheme . '://' . $url->host;
```
Comment (Martchus) [inline script/openqa-clone-job]: I'd prefer using Mojo::URL here as well: `Mojo::URL->new->scheme($url->scheme)->host($url->host)->to_string`

Comment (okurz) [reply script/openqa-clone-job]: I will look into this in another PR crosschecking if there are more occurences than just this single line
